libraries/ts-command-line/src/CommandLineParameter.ts

```diff
@@ -582,12 +591,32 @@ export class CommandLineStringListParameter extends CommandLineParameterWithArgu
         // NOTE: If the environment variable is defined as an empty string,
         // here we will accept the empty string as our value.  (For number/flag we don't do that.)
 
-        // In the current implementation, the environment variable for a "string list" can only
-        // store a single item.  If we wanted to allow multiple items (and still have a conventional-seeming
-        // environment), we would ask the caller to provide an appropriate delimiter.  Getting involved
-        // with escaping here seems unwise, since there are so many shell escaping mechanisms that could
-        // potentially confuse the experience.
-        this._values = [ environmentValue ];
+        if (environmentValue[0] === '[') {
```

Comment:
In Bash, it is easy to assign an empty string to an environment variable:
```sh
export MY_VARIABLE=
```
For the DOS cmd.exe shell, it is possible for the value to be an empty string, but you cannot use SET to assign it -- doing so will delete the variable.  So it has to be assigned using some other tool.  PowerShell can probably do it.

If you're asking about the design here, because an empty string does not start with `[`, it will simply become the first item in the list.

Reply:
I was asking because if `environmentValue` can be `""` then accessing `environmentValue[0]` can throw whenever the string is empty but not undefined. The string can become `""` if it was originally whitespace entirely and then we trimmed the string.

Reply:
Yes in C#, not true in JavaScript.